compiler/qsc_hir/src/hir.rs

```diff
@@ -279,6 +279,53 @@ impl Display for Package {
     }
 }
 
+impl Package {
+    /// Returns a collection of the fully qualified names of any callables annotated with `@Test()`
+    pub fn collect_test_callables(&self) -> std::result::Result<Vec<String>, String> {
+        let items_with_test_attribute = self
+            .items
+            .iter()
+            .filter(|(_, item)| item.attrs.iter().any(|attr| *attr == Attr::Test));
+
+        let callables = items_with_test_attribute
+            .filter(|(_, item)| matches!(item.kind, ItemKind::Callable(_)));
+
+        let callable_names = callables
+        .filter_map(|(_, item)| -> Option<std::result::Result<String, String>>{
+            if let ItemKind::Callable(callable) = &item.kind {
+                if !callable.generics.is_empty() {
+                    return Some(Err(format!("Callable {} has generic type parameters. Test callables cannot have generic type parameters.", callable.name.name)));
+                }
+                if callable.input.kind != PatKind::Tuple(vec![]) {
+                    return Some(Err(format!("Callable {} has input parameters. Test callables cannot have input parameters.", callable.name.name)));
+                }
```

Comment:
now that we have the pass to validate the errors, maybe this function can just silently filter them out? Then this function can return the ones it finds and not need to error out if one is invalid.

Reply:
Elsewhere in the compiler we tend to keep invalid things around for the sake of recovery. If other passes are added after this, in my opinion, we'd probably want them to have access to these callables for better user feedback.